### PR TITLE
Use https://staticcheck.io/ for linting

### LINKED ISSUE #602: Use staticcheck github.com/databrickslabs/terraform-provider-databricks/... instead of abandoned golangci-lint
go-staticheck is integrated with VSCode and seems to be actively supported.
 
https://staticcheck.io/docs#running-the-tools

### PR DESCRIPTION
**Pull request by nfx**
* previous solution, golangci-lint, does no longer seem to be actively maintained
* staticcheck is natively integrated with VSCode IDE
* it's easier to have one common and modern linting framework
* Fixes #602

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -52,7 +52,7 @@ RUN \
     && go get gotest.tools/gotestsum \
     # --> Go symbols and outline for go to symbol support and test support 
     && go get github.com/acroca/go-symbols@v0.1.1 && go get github.com/ramya-rao-a/go-outline@7182a932836a71948db4a81991a494751eccfe77 \
-    # --> GolangCI-lint
-    && go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.25.0 \
+    # --> Static checker
+    && go install honnef.co/go/tools/cmd/staticcheck@latest \
```
Comment (alexott) [inline .devcontainer/Dockerfile:56]: in the push hook we have just `go install honnef.co/go/tools/cmd/staticcheck` -  that this version was reason for failure

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,17 +41,16 @@ Most likely, `terraform init -upgrade -verify-plugins=false -lock=false` would b
...
 ```
 echo "export PATH=\$PATH:$(go env GOPATH)/bin" >> ~/.bash_profile
 ```
 
-Installing `golangci-lint`:
+Installing `staticcheck`:
 ```bash
-curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.27.0
-$(go env GOPATH)/bin/golangci-lint
+go install honnef.co/go/tools/cmd/staticcheck@latest
```
Comment (alexott) [inline CONTRIBUTING.md:53]: same as above, although it works for me on the command-line

**Review by alexott — APPROVED** [review]
overall looks good.

not sure if the `go install honnef.co/go/tools/cmd/staticcheck` vs `go install honnef.co/go/tools/cmd/staticcheck@latest` is important, because the first one fails in command line for me, while second is work. But the first version of PR was failing with 2nd version and works with first one

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (nfx) [reply .devcontainer/Dockerfile:56]: ah! yes, i'll change it in a separate PR

**Pull request merged**